src/tasks/tasks.py

```diff
@@ -1389,6 +1393,86 @@ def collect_metrics(ix_to_tag_dict, tag_group):
         return collected_metrics
 
 
+@register_task("winogender-diagnostic", rel_path="RTE/diagnostics", n_classes=2)
+class WinogenderTask(GLUEDiagnosticTask):
+    """Supported wnogender task """
+
+    def __init__(self, path, max_seq_len, name, n_classes, **kw):
+        super().__init__(path, max_seq_len, name, n_classes, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self.n_classes = n_classes
+
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data = None
+        self.acc_scorer = BooleanAccuracy()
+        self.gender_parity_scorer = GenderParity()
+        self.val_mtric = "%s_accuracy" % name
+
+    def load_data(self):
+        rows = pd.read_json(os.path.join(self.path, "winogender.jsonl"), lines=True)
+        rows["sent1"] = rows["context"].apply(
+            lambda x: process_sentence(self.tokenizer_name, x, self.max_seq_len)
+        )
+        rows["sent2"] = rows["hypothesis"].apply(
+            lambda x: process_sentence(self.tokenizer_name, x, self.max_seq_len)
+        )
+        data = list(rows.T.to_dict().values())
+        self.train_data_text = data
+        self.val_data_text = data
+        self.test_data_text = data
+        self.sentences = [x["sent1"] for x in self.train_data_text]
+        self.sentences += [x["sent2"] for x in self.train_data_text]
+
+    def process_split(self, split, indexers):
+        is_using_bert = "bert_wpm_pretokenized" in indexers
+
+        def _make_instance(record):
+            """ from multiple types in one column create multiple fields """
+            d = {}
+            if is_using_bert:
+                input_final = record["sent1"] + record["sent2"][1:]
+                d["inputs"] = sentence_to_text_field(input_final, indexers)
+            else:
+                d["input1"] = sentence_to_text_field(record["sent1"], indexers)
+                d["input2"] = sentence_to_text_field(record["sent2"], indexers)
+            d["sent1_str"] = MetadataField(record["sent1"])
+            d["idx"] = LabelField(int(record["index"]), label_namespace="idxs", skip_indexing=True)
+            d["pair_id"] = LabelField(
+                record["pair-id"], label_namespace="pair_id", skip_indexing=True
+            )
+            d["sent2_str"] = MetadataField(record["sent2"])
+            d["labels"] = LabelField(
+                int(record["label"]), label_namespace="labels", skip_indexing=True
+            )
+            return Instance(d)
+
+        instances = map(_make_instance, split)
+        return instances
+
+    def get_metrics(self, reset=False):
+        return {
+            "accuracy": self.acc_scorer.get_metric(reset),
+            "gender_parity": self.gender_parity_scorer.get_metric(reset),
+        }
+
+    def update_diagnostic_metrics(self, logits, labels, batch):
```

Comment:
where is this getting called?

Reply:
This gets called at src/models.py instead of update_metrics() for normal non-diagnostic tasks.